src/lib/shopify/storefront/provider.tsx

```diff
@@ -33,6 +34,7 @@ type Context = {
   cartItemsCount: number | undefined
   cartToggleState: ToggleState
   errors: TErrors
+  events: EventEmitter
```

Comment:
This typing is a generic one for `EventEmmiter`, but would it be possible to type it with our custom events? So that TypeScript expects a `createCartSuccess`, etc.